rust/cargo-psibase/src/link.rs

```diff
@@ -291,6 +297,26 @@ fn provide_dest_seq_id(source_block_id : InstrSeqId, source_function : &LocalFun
                 dest_seq.instr(i);
             }
 
+            // Local reference updates
+            Instr::LocalGet(local_ref) => {
+                let mut i = local_ref.clone();
+                let ty = source.locals.get(local_ref.local).ty();
+                i.local = dest.locals.add(ty);
```

Comment:
Are you sure that this is right? It looks to me like it's creating a new variable for every instruction

Reply:
Def would not say I'm sure it's right. But I do know that adding the local will only actually add something if it's a new type. And also the locals object here is internal to walrus, once you actually emit the wasm there's no actual "locals" table

Reply:
That would also be wrong as it would merge all locals of the same type. According to https://docs.rs/walrus/latest/walrus/struct.ModuleLocals.html#method.add, it always returns a new local id. Also, I don't quite understand your comment: it's the ID of the local, not the ID of a type.

Reply:
Meant to say that adding a `type` would only add one if it's a new type - regardless I guess not really relevant. 

Okay I updated this but I'm still not sure it's correct. In the latest, it should create a new local only if it's the first time a local with a given ID is referenced while parsing instructions from the source module. Because I'm not seeing an instruction that explicitly creates a local?

Reply:
There isn't an instruction that creates locals. Locals are defined in the function prologue.  Function parameters are accessed as locals, so they also need to be translated.